src/analysis/processing/qgsalgorithmexportlayersinformation.cpp

```diff
@@ -0,0 +1,174 @@
+  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ), QObject::tr( "Output" ), QgsProcessing::TypeVectorPolygon, QVariant() ) );
+}
+
+QString QgsExportLayersInformationAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Creates a polygon layer with features corresponding to the extent of selected layer(s).\n\n"
+                      "Additional layer details - CRS, provider name, file path, layer name, subset filer, abstract and attribution - are attached as attributes to each feature." );
+}
+
+QgsExportLayersInformationAlgorithm *QgsExportLayersInformationAlgorithm::createInstance() const
+{
+  return new QgsExportLayersInformationAlgorithm();
+}
+
+bool QgsExportLayersInformationAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  const QList< QgsMapLayer * > layers = parameterAsLayerList( parameters, QStringLiteral( "LAYERS" ), context );
+  for ( QgsMapLayer *layer : layers )
+  {
+    if ( !mCrs.isValid() )
+    {
+      mCrs = layer->crs();
+    }
+    else if ( mCrs.authid() != QStringLiteral( "EPSG:4326" ) )
+    {
+      if ( mCrs != layer->crs() )
+      {
+        // mixed CRSes, set output CRS to EPSG:4326
+        mCrs = QgsCoordinateReferenceSystem( QStringLiteral( "EPSG:4326" ) );
+      }
+    }
+    mLayers.emplace_back( layer->clone() );
+  }
+
+  if ( !mCrs.isValid() )
+    mCrs = QgsCoordinateReferenceSystem( QStringLiteral( "EPSG:4326" ) );
+
+  if ( mLayers.empty() )
+    feedback->reportError( QObject::tr( "No layers selected" ), false );
+
+  return true;
+}
+
+QVariantMap QgsExportLayersInformationAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  QgsFields outFields;
+  outFields.append( QgsField( QStringLiteral( "name" ), QVariant::String ) );
+  outFields.append( QgsField( QStringLiteral( "source" ), QVariant::String ) );
+  outFields.append( QgsField( QStringLiteral( "crs" ), QVariant::String ) );
+  outFields.append( QgsField( QStringLiteral( "provider" ), QVariant::String ) );
+  outFields.append( QgsField( QStringLiteral( "file_path" ), QVariant::String ) );
+  outFields.append( QgsField( QStringLiteral( "layer_name" ), QVariant::String ) );
+  outFields.append( QgsField( QStringLiteral( "subset" ), QVariant::String ) );
+  outFields.append( QgsField( QStringLiteral( "abstract" ), QVariant::String ) );
+  outFields.append( QgsField( QStringLiteral( "attribution" ), QVariant::String ) );
+
+  QString outputDest;
+  std::unique_ptr< QgsFeatureSink > outputSink( parameterAsSink( parameters, QStringLiteral( "OUTPUT" ), context, outputDest, outFields,
+      QgsWkbTypes::Polygon, mCrs ) );
+
+  const QList< QgsMapLayer * > layers = parameterAsLayerList( parameters, QStringLiteral( "LAYERS" ), context );
+
+  double step = layers.size() > 0 ? 100.0 / layers.size() : 1;
+  int i = 0;
+  for ( const auto &layer : mLayers )
```

Comment:
```suggestion
  for ( const QgsMapLayer* layer : mLayers )
```

Reply:
auto here is std::unique_ptr, I'll update accordingly. Goes to show how misleading auto is.